lightning/src/ln/channel.rs

```diff
@@ -251,76 +251,293 @@ enum HTLCUpdateAwaitingACK {
-	/// and our counterparty consider the funding transaction confirmed.
-	FundingSent = 8,
-	/// Flag which can be set on `FundingSent` to indicate they sent us a `channel_ready` message.
-	/// Once both `TheirChannelReady` and `OurChannelReady` are set, state moves on to `ChannelReady`.
-	TheirChannelReady = 1 << 4,
-	/// Flag which can be set on `FundingSent` to indicate we sent them a `channel_ready` message.
-	/// Once both `TheirChannelReady` and `OurChannelReady` are set, state moves on to `ChannelReady`.
-	OurChannelReady = 1 << 5,
-	ChannelReady = 64,
-	/// Flag which is set on `ChannelReady` and `FundingSent` indicating remote side is considered
-	/// "disconnected" and no updates are allowed until after we've done a `channel_reestablish`
-	/// dance.
-	PeerDisconnected = 1 << 7,
-	/// Flag which is set on `ChannelReady`, FundingCreated, and `FundingSent` indicating the user has
-	/// told us a `ChannelMonitor` update is pending async persistence somewhere and we should pause
-	/// sending any outbound messages until they've managed to finish.
-	MonitorUpdateInProgress = 1 << 8,
-	/// Flag which implies that we have sent a commitment_signed but are awaiting the responding
-	/// revoke_and_ack message. During this time period, we can't generate new commitment_signed
-	/// messages as then we will be unable to determine which HTLCs they included in their
-	/// revoke_and_ack implicit ACK, so instead we have to hold them away temporarily to be sent
-	/// later.
-	/// Flag is set on `ChannelReady`.
-	AwaitingRemoteRevoke = 1 << 9,
-	/// Flag which is set on `ChannelReady` or `FundingSent` after receiving a shutdown message from
-	/// the remote end. If set, they may not add any new HTLCs to the channel, and we are expected
-	/// to respond with our own shutdown message when possible.
-	RemoteShutdownSent = 1 << 10,
-	/// Flag which is set on `ChannelReady` or `FundingSent` after sending a shutdown message. At this
-	/// point, we may not add any new HTLCs to the channel.
-	LocalShutdownSent = 1 << 11,
-	/// We've successfully negotiated a closing_signed dance. At this point ChannelManager is about
-	/// to drop us, but we store this anyway.
-	ShutdownComplete = 4096,
-	/// Flag which is set on `FundingSent` to indicate this channel is funded in a batch and the
-	/// broadcasting of the funding transaction is being held until all channels in the batch
-	/// have received funding_signed and have their monitors persisted.
-	WaitingForBatch = 1 << 13,
+	/// We are negotiating the parameters required for the channel prior to funding it.
+	NegotiatingFunding(NegotiatingFundingFlags),
+	/// We have sent `funding_created` and are awaiting a `funding_signed` to advance to
+	/// `FundingSent`. Note that this is nonsense for an inbound channel as we immediately generate
+	/// `funding_signed` upon receipt of `funding_created`, so simply skip this state.
+	FundingCreated,
```

Comment:
I guess this could be used for both inbound and outbound channels in V2 establishment when the one party has sent the initial `commitment_signed` and not received the counterparty`s yet. I don't know what a good general name for that would be but happy to just use this state.